### PR TITLE
Testing TicTacToe methods for review.

### PR DESCRIPTION
**Pull request by CalyCherkaoui**
This pull-request is related to Testing Tic-Tac-Toe game methods.
The methods tested are : 
-  winner?
- grid_filled?
- pick_valid?
- not_picked?
- update_grid

PS : we struggled to configure the GitHub linter. Any advice will be welcome . Thanks

### CODE REVIEW

File: spec/project_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# spec/project_spec.rb
+require './lib/player'
+require './lib/board'
```
Comment (suretrust) [inline spec/project_spec.rb]: `require` should be used for external files, like gems, while `require_relative` should be used for referring to files within your directory.

File: spec/project_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  end
+end
+
+describe Board do
+  let(:test_board) { Board.new([1, 2, 'X']) }
+  describe '#grid_filled?' do
+    it 'grid is not completely filled with strings' do
+      expect(test_board.grid_filled?).not_to eql(true)
+    end
+  end
```
Comment (suretrust) [inline spec/project_spec.rb]: - [ ] Your specs should cover positive and negative scenarios. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/)).

> In this example, you need to test with two arrays, one that's filled another that's not completely filled, then test as shown below (this is just an example implementing the positive and negative scenarios, you can do it anyhow you see fit).

```suggestion
  describe '#grid_filled?' do
    it 'returns true if grid is completely filled with strings' do
      expect(filled_test_board.grid_filled?).to eq(true)
    end
    
    it 'returns false if grid is not completely filled with strings' do
      expect(unfilled_test_board.grid_filled?).to eq(false)
    end
  end
```

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi team, 👋 

### Great job thus far! 🙌 Please see my comments. 👇 

- [x] Add linter to your repo. You could check the [troubleshooting section of the linter guide repo](https://github.com/microverseinc/linters-config/blob/master/README.md#troubleshooting) to see if you're missing anything.

- [x] Add positive and negative scenarios for all tests as illustrated below.

**Kindly submit another CRR once these issues have been resolved.** _Happy coding!_ 🎉

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team      
#### Great Job so far, but there are a few more things required. 
### General
- [x] Lets write test for the initialize method in `Game` class
- [x] Lets test for  a draw scenario between players.
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by suretrust — APPROVED** [review]
Hi team, 👋 

### Great job implementing the suggested changes. 💯 Let's get this merged! 👍 

- [x] You could check [better specs](http://www.betterspecs.org/) in your free time to check best `rspec` practices.

_Good luck with your future projects and happy coding!_ 🎉
